### PR TITLE
Replanning - 02/01

### PR DESCRIPTION
**Pull request by aravindsiv**
Some changes based on our discussion with Kostas last week. Specifically:
* Replanner client will now terminate if the current observation is inside the goal region (but the MuSHR might still have some controls that it will continue executing, bringing it outside the goal region. So, if you have any ideas on how we can override this behavior, I can implement it)
* Replace the old interface with nodelets
* Not working with the camera since I have issues with the RGB image when logging in remotely. I also need to figure out Aruco detection and camera extrinsic parameters.

WIP: 
* On `ML4KP-devel`, define two-way obstacle translation from Mujoco XML and ML4KP YAML.
* Generate statistics on planning errors.

### CODE REVIEW

File: motion_planning/include/motion_planning/planner_client.hpp [hunk 1]
```diff
@@ -23,7 +23,7 @@ class planner_client_t
     const std::string service_name{ root + "/planner_service" };
     _service_client = nh.serviceClient<Service>(service_name);
     _obs_subscriber = nh.subscribe(root + "/pose", 1000, &planner_client_t::observation_callback, this);
-    _plan_publisher = nh.advertise<ml4kp_bridge::Plan>(root + "/ml4kp_plan", 1000, true);
+    _plan_publisher = nh.advertise<ml4kp_bridge::PlanStamped>(root + "/ml4kp_plan", 1000, true);
```
Comment (garygra) [inline motion_planning/include/motion_planning/planner_client.hpp:26]: Is there a reason to have `1000`? I have been using `1` (my understanding is that it has an impact on memory)

File: motion_planning/include/motion_planning/planner_client.hpp [hunk 2]
```diff
@@ -51,23 +63,22 @@ class planner_client_t
...
     _service.request.goal_configuration = goal_configuration;
     _service.request.return_trajectory.data = _publish_trajectory;
     if (_service_client.call(_service))
     {
       ROS_DEBUG("Service call successful");
-      if (_service.response.planner_output == Service::Response::TYPE_SUCCESS)
+      if (is_goal_reached(goal_configuration, goal_radius))
+      {
+        ROS_INFO("Goal reached. Not publishing plan");
+        // TODO: Check if zero control has been published last
```
Comment (garygra) [inline motion_planning/include/motion_planning/planner_client.hpp:78]: I think having zero control is not the based idea (at some point we talked about having a controller like lqr). Maybe at this point we call a controller, which for now can be a "set zero" one.

File: task_planning/src/mushr_replanning.cpp [hunk 3]
```diff
@@ -98,8 +98,16 @@ int main(int argc, char** argv)
...
-    planner_client.call_service(goal_configuration, goal_radius, planning_cycle_duration, first_cycle);
-    first_cycle = false;
+    if (!planner_client.is_goal_reached(goal_configuration, goal_radius))
+    {
+      planner_client.call_service(goal_configuration, goal_radius, planning_cycle_duration, first_cycle);
+      first_cycle = false;
+    }
+    else
+    {
+      ROS_WARN("Goal reached, not replanning");
```
Comment (garygra) [inline task_planning/src/mushr_replanning.cpp:108]: I guess the controller should go here

**Review by garygra — APPROVED** [review]
Good

File: motion_planning/include/motion_planning/planner_client.hpp (hunk 2 above)
Comment (aravindsiv) [reply motion_planning/include/motion_planning/planner_client.hpp:78]: Will get to this as part of a future PR.

**Pull request merged**